lua/obsidian/commands/extract_note.lua

```diff
@@ -0,0 +1,37 @@
+local log = require "obsidian.log"
+local util = require "obsidian.util"
+
+---Extract the selected text into a new note
+---and replace the selection with a link to the new note.
+---
+---@param client obsidian.Client
+return function(client, data)
+  local viz = util.get_visual_selection()
+  if viz.lines == nil or #viz.lines == 0 then
+    log.err "ObsidianExtractNote must be called with visual selection"
+    return
+  end
+
+  local content
+  if data.args ~= nil and string.len(data.args) > 0 then
+    content = { data.args }
+  else
+    content = viz.lines
+  end
+
+  -- new note with title from user
+  local title = vim.fn.input { prompt = "Enter title (optional): " }
+  if string.len(title) == 0 then
+    title = nil
+  end
+  local note = client:new_note(title)
+
+  -- replace selection with link to new note
+  local link = client:format_link(note)
+  vim.api.nvim_buf_set_lines(0, viz.csrow - 1, viz.cerow + 1, false, { link })
```

Comment:
I think this should be:
```suggestion
  vim.api.nvim_buf_set_text(0, viz.csrow - 1, viz.cscol - 1, viz.cerow - 1, viz.cecol, { link })
```

That way you keep the rest of the start line / end line intact.

Reply:
Good catch.